lib/std/crypto/kyber_d00.zig

```diff
@@ -1021,7 +1021,7 @@ const Poly = struct {
                 //                  = ⌊((x << d) + q/2) / q⌋ mod⁺ 2ᵈ
                 //                  = DIV((x << d) + q/2, q) & ((1<<d) - 1)
                 const t = @intCast(u32, p.cs[in_off + i]) << d;
-                in[i] = @intCast(u16, @divFloor(t + q_over_2, Q) & two_d_min_1);
+                in[i] = @intCast(u16, @divFloor(t + q_over_2, @as(u32, Q)) & two_d_min_1);
```

Comment:
what happened here? can you please add a compile error test case for what happened before you added `@as` to resolve the issue here?

Reply:
This and the other similar changes have been reverted because I made a change to mirror existing semantics for now. I think said semantics are quite poor and should be changed, but I'll open a proposal to track that.